### PR TITLE
consider GracePeriod contracts in nodes' used resources

### LINKED ISSUE #172: Field to deploy on 'node 11' on 'devnet'
node11issue

### Description
This happened when I tried to deploy on node 11, I used another node and works well

### PR DESCRIPTION
**Pull request by mariobassem**

### CODE REVIEW

File: internal/explorer/db/postgres.go [hunk 1]
```diff
@@ -51,7 +51,7 @@ const (
 		COALESCE(node_resources_total.sru, 0) as total_sru
 	FROM contract_resources
 	JOIN node_contract as node_contract
-	ON node_contract.resources_used_id = contract_resources.id AND node_contract.state = 'Created'
+	ON node_contract.resources_used_id = contract_resources.id AND node_contract.state IN ('Created')
```
Comment (AhmedHanafy725) [inline internal/explorer/db/postgres.go]: missing grace period?

**Review by AhmedHanafy725 — COMMENTED** [review]
please add the issue related to the PR's description

File: tools/db/generate.go [hunk 2]
```diff
@@ -125,8 +125,15 @@ func generateContracts(db *sql.DB) error {
...
-			state = "Created"
+		if nodeUP[nodeID] {
+			if flip(contractCreatedRatio) {
+				state = "Created"
+			} else if flip(0.05) {
+				state = "GracePeriod"
+			}
+		}
+		if state != "Deleted" && (minContractHRU > nodesHRU[nodeID] || minContractMRU > nodesMRU[nodeID] || minContractSRU > nodesSRU[nodeID]) {
+			continue
```
Comment (OmarElawady) [inline tools/db/generate.go:136]: can you add i-- (to maintain the number of contracts to be equal to contractsCount)?

Comment (OmarElawady) [inline tools/db/generate.go:137]: can you add i-- (to maintain the number of contracts to be equal to contractsCount)?

**Review by OmarElawady — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**